packages/react-native-otlp/android/src/main/java/io/embrace/rnembraceotlp/RNEmbraceOTLPModule.kt

```diff
@@ -1,50 +1,158 @@
 package io.embrace.rnembraceotlp
 
-import android.util.Log
+import java.time.Duration
+import java.util.logging.Logger
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+import kotlin.time.toJavaDuration
+
 import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.bridge.ReactContextBaseJavaModule
 import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.ReadableArray
 import com.facebook.react.bridge.ReadableMap
 
 import io.embrace.android.embracesdk.Embrace
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter
+import io.opentelemetry.exporter.otlp.http.logs.OtlpHttpLogRecordExporter
+
+data class HeaderConfig (
+    val key: String,
+    val token: String
+)
+
+data class ExporterConfig(
+    val endpoint: String,
+    val headers: List<HeaderConfig>? = emptyList(),
+    val timeout: Duration? = null
+)
+
+data class OtlpExporterConfig (
+    val traceExporter: ExporterConfig? = null,
+    val logExporter: ExporterConfig? = null
+)
+
+// parsing Readable Map into the real OTLP Exporter Config shape
+fun parseExportConfig(exportConfig: ReadableMap): OtlpExporterConfig {
+    var spanExportConfigParsed: ExporterConfig? = null
+    val spanExportConfig = exportConfig.getMap("traceExporter")
+    val spanExportEndpoint = spanExportConfig?.getString("endpoint")
+
+    if (spanExportConfig != null && !spanExportEndpoint.isNullOrBlank()) {
+        val headersExportConfig = spanExportConfig.getArray("headers")?.let {
+            parseHeaders(it)
+        }
+
+        val timeoutExportConfig = parseTimeout(spanExportConfig.getDouble("timeout"))
+
+        spanExportConfigParsed = ExporterConfig(
+            endpoint = spanExportEndpoint,
+            headers = headersExportConfig,
+            timeout = timeoutExportConfig,
+        )
+    }
+
+    return OtlpExporterConfig(traceExporter = spanExportConfigParsed)
+}
+
+fun parseHeaders(headers: ReadableArray): List<HeaderConfig> {
+    val headerList: MutableList<HeaderConfig> = arrayListOf()
+
+    for (i in 0 until headers.size()) {
+        val header = headers.getMap(i)
+        val keyVal = header.getString("key")
+        val tokenVal = header.getString("token")
+
+        if (!keyVal.isNullOrBlank() && !tokenVal.isNullOrBlank()) {
+            headerList.add(HeaderConfig(key = keyVal, token = tokenVal))
+        }
```

Comment:
maybe logging a warning here if an invalid header was passed missing one of these